src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx

```diff
@@ -44,16 +41,26 @@ const ProfilePictureInput: React.FunctionComponent<ProfilePictureInputProps> = (
   const [error, setError] = useState("");
   const firebase = useFirebase();
   const uploadRef = useRef<HTMLInputElement>(null);
-  const queryParams = useQuery();
-  const venueId = queryParams.get("venueId");
 
-  const customAvatars = useAsync(async () => {
+  // @debt Replace fetchSovereignVenueId with useSovereignVenueId, when the hook is refactored to accept venueId as a param.
+  const {
+    value: sovereignVenueId,
+    loading: isLoadingSovereignVenueId,
+  } = useAsync(async () => await fetchSovereignVenueId(venueId));
+
+  const {
+    value: customAvatars,
+    loading: isLoadingCustomAvatars,
+  } = useAsync(async () => {
     const storageRef = firebase.storage().ref();
-    const list = await storageRef.child(`/assets/avatars/${venueId}`).listAll();
-    const promises = list.items.map((item) => item.getDownloadURL());
-    const avatars: string[] = await Promise.all(promises);
+    const list = await storageRef
+      .child(`/assets/avatars/${sovereignVenueId}`)
+      .listAll();
+    const avatars: string[] = await Promise.all(
+      list.items.map((item) => item.getDownloadURL())
+    );
     return avatars;
-  }, []);
+  }, [sovereignVenueId]);
```

Comment:
[NB] Still not sure if this pattern of listing/reading the storage like this is a good idea.. I'm surprised it even works without changing our security rules.

Reply:
There is a read rule which allows us to read `assets/`

Reply:
You can take a look at `storage.rules`

Reply:
For the admin we'd have to add write and delete with some conditions ofc.

Reply:
Mm, fair. I didn't realise that would also let us list assets/etc too. Makes sense though.